### PR TITLE
Packed Sprites

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
Sprites of different sizes can be packed on a sprite sheet. Data format used is from JSON(hash) of TexturePacker (TP).

SpriteManager works as before all spritesizes are the same, cell data determined from cellIndex.
SpritePackedManager requires image file _path/name.imgext_ and JSON file _path/name.json_ in same folder with same name different extensions, cell data read from JSON file or image file plus JSON object.

| Testing | Test Url |
|-----|-----|
|Backward Compatability | https://tender-murdock-275855.netlify.com/sprites/index.html |
| Direct JSON TP format cellRef | https://tender-murdock-275855.netlify.com/sprites/test0.html |
|Direct JSON min format cellRef | https://tender-murdock-275855.netlify.com/sprites/test6.html |
|Direct JSON min format cellIndex | https://tender-murdock-275855.netlify.com/sprites/test2.html |
| File JSON cellRef | https://tender-murdock-275855.netlify.com/sprites/test3.html |
| File JSON cellIndex | https://tender-murdock-275855.netlify.com/sprites/test4.html |
| File JSON animation | https://tender-murdock-275855.netlify.com/sprites/test5.html |
| File JSON multiple sprites | https://tender-murdock-275855.netlify.com/sprites/test7.html |

Will work on documentation when and if approved.

### CODE REVIEW

**Comment by deltakosh** [comment]
Oh btw you just have a doc issue to fix

File: src/Sprites/spritePackedManager.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * @param imgUrl defines the sprite sheet url
+     * @param capacity defines the maximum allowed number of sprites
+     * @param scene defines the hosting scene
+     * @param spriteJSON, null otherwise a JSON object defining sprite sheet data
+     * @param epsilon defines the epsilon value to align texture (0.01 by default)
+     * @param samplingMode defines the smapling mode to use with spritesheet
+     * @param fromPacked set to true; do not alter
+     */
+
+    constructor(public name: string, imgUrl: string, capacity: number, scene: Scene, spriteJSON: string | null = null, epsilon: number = 0.01, samplingMode: number = Texture.TRILINEAR_SAMPLINGMODE) {
```
Comment (deltakosh) [inline src/Sprites/spritePackedManager.ts]: Even if this is painful you need to also add comments for each public member INSIDE the constructor.
something like:
```
constructor(
/** blah blah blah */
public name: string
```

Comment (BabylonJSGuide) [reply src/Sprites/spritePackedManager.ts]: Done. Started works on docs.

Comment (BabylonJSGuide) [reply src/Sprites/spritePackedManager.ts]: Do not understand the new documentation error on spriteJSON. Please advise.

File: src/Sprites/spriteManager.ts [hunk 2]
```diff
@@ -131,15 +140,18 @@ export class SpriteManager implements ISpriteManager {
      * @param scene defines the hosting scene
      * @param epsilon defines the epsilon value to align texture (0.01 by default)
      * @param samplingMode defines the smapling mode to use with spritesheet
+     * @param fromPacked set to false; do not alter
+     * @param spriteJSON, null otherwise a JSON object defining sprite sheet data; do not alter
```
Comment (deltakosh) [inline src/Sprites/spriteManager.ts]: You need to remove the comma after spriteJSON I guess

**Comment by BabylonJSGuide** [comment]
Yay all checks passed :)

**Pull request merged**